### PR TITLE
Split RewardEscrowV2 contract in L1 and L2 counterpart

### PR DESCRIPTION
**Pull request by i-stam**
Former RewardEscrowV2 contract is split in two:
- RewardEscrowV2 residing on L1.
- OptimisticRewardEscrowV2 residing on L2.
- Both inherit from BaseRewardEscrowV2 the basic functionality. 

The L1/L2 specific functions revert in the base contract.

### CODE REVIEW

File: contracts/OptimisticRewardEscrowV2.sol [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+// https://docs.synthetix.io/contracts/RewardEscrow
+contract OptimisticRewardEscrowV2 is BaseRewardEscrowV2 {
+    /* ========== ADDRESS RESOLVER CONFIGURATION ========== */
+    bytes32 private constant CONTRACT_SYNTHETIX_BRIDGE_BASE = "SynthetixBridgeToBase";
+
+    /* ========== CONSTRUCTOR ========== */
+
+    constructor(address _owner, address _resolver) public BaseRewardEscrowV2(_owner, _resolver) {
+        appendToAddressCache(CONTRACT_SYNTHETIX_BRIDGE_BASE);
```
Comment (jacko125) [inline contracts/OptimisticRewardEscrowV2.sol:16]: This is good to split the address to cache only specific for L2.

File: contracts/interfaces/IRewardEscrowV2.sol [hunk 2]
```diff
@@ -2,15 +2,18 @@ pragma solidity >=0.4.24;
 pragma experimental ABIEncoderV2;
 
 
-interface IRewardEscrowV2 {
+library VestingEntries {
```
Comment (jacko125) [inline contracts/interfaces/IRewardEscrowV2.sol:5]: Good idea to move this here into the interface. 

We've also previously just kept it inside the `interface IRewardEscrowV2` as a struct

**Review by jacko125 — APPROVED** [review]
lgtm! great effort splitting these out, will be really helpful.

File: contracts/OptimisticRewardEscrowV2.sol [hunk 3]
```diff
@@ -0,0 +1,51 @@
+pragma solidity ^0.5.16;
+pragma experimental ABIEncoderV2;
+
+// Inheritance
+import "./BaseRewardEscrowV2.sol";
+
+
+// https://docs.synthetix.io/contracts/RewardEscrow
+contract OptimisticRewardEscrowV2 is BaseRewardEscrowV2 {
```
Comment (justinjmoses) [inline contracts/OptimisticRewardEscrowV2.sol:9]: I think for naming conventions we call this `ImportableRewardEscrowV2`

File: contracts/interfaces/IRewardEscrowV2.sol (hunk 2 above)
Comment (i-stam) [reply contracts/interfaces/IRewardEscrowV2.sol:5]: I was actually getting an error. Unfortunately, interface inheritance is not supported in the current version. We had to declare it inside the interface to be able to use it there as a return a parameter but the compiler was complaining about the inheriting contract not declaring. Using a library was the easiest solution I came up with.

File: contracts/OptimisticRewardEscrowV2.sol (hunk 3 above)
Comment (i-stam) [reply contracts/OptimisticRewardEscrowV2.sol:9]: Yeap, I wanted to ask about opinions on naming. I like that `ImportableRewardEscrowV2` follows the same pattern as BaseSynthetix-MintableSynthetix.

File: contracts/RewardEscrow.sol [hunk 4]
```diff
@@ -1,4 +1,5 @@
 pragma solidity ^0.5.16;
+pragma experimental ABIEncoderV2;
```
Comment (jacko125) [inline contracts/RewardEscrow.sol:2]: @i-stam  We don't need to update the old RewardEscrow contract with the `pragma experimental ABIEncoderV2`

Comment (i-stam) [reply contracts/RewardEscrow.sol]: You are right, I was testing something and forgot tor remove it. Fixed.

**Pull request merged**